src/equations/ideal_glm_mhd_2d.jl

```diff
@@ -1075,8 +1083,34 @@ end
   return c_f
 end
 
+@inline function calc_fast_wavespeed(cons, normal_direction::AbstractVector, equations::IdealGlmMhdEquations2D)
+  rho, rho_v1, rho_v2, rho_v3, rho_e, B1, B2, B3, psi = cons
+  v1 = rho_v1 / rho
+  v2 = rho_v2 / rho
+  v3 = rho_v3 / rho
+  kin_en = 0.5 * (rho_v1 * v1 + rho_v2 * v2 + rho_v3 * v3)
+  mag_en = 0.5 * (B1 * B1 + B2 * B2 + B3 * B3)
+  p = (equations.gamma - 1) * (rho_e - kin_en - mag_en - 0.5 * psi^2)
+  a_square = equations.gamma * p / rho
+  sqrt_rho = sqrt(rho)
+  b1 = B1 / sqrt_rho
+  b2 = B2 / sqrt_rho
+  b3 = B3 / sqrt_rho
+  b_square = b1^2 + b2^2 + b3^2
```

Comment:
Now I see this too. what about `b1 * b1 + b2 * b2 + b3 * b3` ? Or do we not want to go overboard avoiding powers?

Reply:
Done in e483df3b